### ISSUE TITLE
AutoReloading does not work

### ISSUE THREAD

**Issue by AdityaAnand1** [ktorio/ktor#1600]
The example is not very clear, adding the packagename from file heirarchy: `src/<packagename>` does not throw any error, but neither does it work.

```
 watchPaths = listOf("mypackage")
```

What exactly is the correct syntax? I really just want a wildcard. Anything in my project changes, go reload.

Also, is there logging when auto-reload will trigger?

**Comment by cy6erGn0m** [comment]
https://ktor.io/servers/autoreload.html

**Comment by AdityaAnand1** [comment]
Perhaps I (and several others from what I gather from other issues like this) are missing some key step or understanding as even after following the documentation perfectly, auto-reloading still does not work.

My suggestion: Create a small sample project that can be merely cloned and run to demonstrate the auto-reload feature. Perhaps add in a script that runs any required commands along with the project.

That should quell such issues once and for all 😃

**Comment by cy6erGn0m** [comment]
Ah, most likely the problem is that the package name is not a part of classpath. Watch patterns are applied to a root classpaths list. I would try to add app/module name instead.

**Comment by dvdgsng** [comment]
Having the same issue. My setup is already based on the auto-reload documentation. See error below.

Auto-Reload via `gradle -t installDist` + `gradle run` works, however.

ktor_version 1.4.0
openjdk 14.0.1
IntelliJ IDEA 2020.2.1 Build #IU-202.6948.69

```
package com.example

fun Application.module() {
   ... do stuff ...
}

fun main(args: Array<String>) {
    println(System.getProperty("java.class.path").contains("foobar")) // <- is true

    embeddedServer( Netty, watchPaths = listOf("foobar"), port = 8080,
        module = Application::module
    ).apply { start(wait = true) }
}
```

```
ktor {
    deployment {
        host = localhost
        port = 8080
        watch = [ foobar ]
    }
    application {
        modules = [ com.example.ApplicationKt.module ]
    }
}
```



```
Exception in thread "main" java.lang.RuntimeException: Module function provided as lambda cannot be unlinked for reload
	at io.ktor.server.engine.ApplicationEngineEnvironmentReloading.<init>(ApplicationEngineEnvironmentReloading.kt:58)
	at io.ktor.server.engine.ApplicationEngineEnvironmentBuilder.build(ApplicationEngineEnvironment.kt:105)
	at io.ktor.server.engine.ApplicationEngineEnvironmentKt.applicationEngineEnvironment(ApplicationEngineEnvironment.kt:44)
	at io.ktor.server.engine.EmbeddedServerKt.embeddedServer(EmbeddedServer.kt:55)
	at io.ktor.server.engine.EmbeddedServerKt.embeddedServer(EmbeddedServer.kt:36)
	at io.ktor.server.engine.EmbeddedServerKt.embeddedServer$default(EmbeddedServer.kt:34)
	at com.example.ApplicationKt.main(Application.kt:102)
```

**Comment by e5l** [comment]
Fixed with 1.5.3
